### PR TITLE
Vendorize skimage

### PR DESCRIPTION
**Pull request by nbren12**
Scikit-image is a difficult dependency to manage (especially since it makes strong assumptions about pooch). Even an import of scikit-image mutates the users ~/.cache folder, which can lead to hard to debug permissions errors and other issues. We work around this via careful dependency pinning, but I still frequently run into this problem when e.g. updating versions. We only use two small functions from scikit-image, so it's easy to vendorize them in this repo.

We can reimplement this using `xarray.coarsen` at another time (see #1048).

Added public API:
- vcm.testing.checksum_dataarray
- vcm.testing.checksum_dataarray_mapping

Significant internal changes:
- rewired the coarening tests to use regression checksums instead.

Requirement changes:
- removed scikit-image
- I would've liked to remove pooch too, but metpy uses it too, albeit only in one module that we probably don't import.

### CODE REVIEW

File: external/vcm/tests/test_cubedsphere.py [hunk 1]
```diff
@@ -260,27 +260,16 @@ def input_dataset(input_dataarray):
...
+):
     if use_dask:
         input_dataarray = input_dataarray.chunk({"x": 2, "y": 2, "z": -1})
-
     block_sizes = {"x": 2, "y": 2}
     result = _xarray_block_reduce_dataarray(
         input_dataarray, block_sizes, reduction_function
     )
-    assert_identical_including_dtype(result, expected)
+    print(vcm.testing.checksum_dataarray(result), file=regtest)
```
Comment (spencerkclark) [inline external/vcm/tests/test_cubedsphere.py:272]: It looks like `checksum_dataarray` only takes the hash of the values of the DataArray (and ignores coordinate values and attributes).  Propagating attributes in particular was an issue with the coarsening functions in the past (#513), so I think we may want to revert this change (or modify the way `checksum_dataarray` operates).

**Review by spencerkclark — COMMENTED** [review]
It's great this was so simple to vendor over.  I just have a minor comment about the modification to the test of `_xarray_block_reduce_dataarray`.

File: external/vcm/tests/test_cubedsphere.py (hunk 1 above)
Comment (nbren12) [reply external/vcm/tests/test_cubedsphere.py:272]: yah i can do that.

**Review by spencerkclark — APPROVED** [review]
